test/webjobs.js

```diff
@@ -0,0 +1,452 @@
+"use strict";
+
+var assert = require("assert");
+var fs = require("fs");
+var JSZip = require("jszip");
+var retry = require("retry");
+var api = require("../")({website: process.env.WEBSITE, username: process.env.USERNAME, password: process.env.PASSWORD});
+
+var triggeredFiles = {
+    "run.cmd": "echo hello world %*"
+};
+var continuousFiles = {
+    "run.js": "console.log(process.argv.join(' ')); setInterval(() => console.log('ping'), 30000);"
+};
+
+function createJobZip(localPath, files, cb) {
+    var generateOptions = {
+        type: "nodebuffer",
+        streamFiles: true
+    };
+
+    var zip = new JSZip();
+
+    Object.keys(files).forEach(function (key) {
+        zip.file(key, files[key]);
+    });
+
+    zip.generateNodeStream(generateOptions)
+        .pipe(fs.createWriteStream(localPath))
+        .on("error", cb)
+        .on("finish", cb);
+}
+
+function createPollingCallback(cb) {
+    var operation = retry.operation({
+        retries: 5
+    });
```

Comment:
Might seem like overkill, but I really wanted to make sure these tests run first-time-every-time. Hopefully this eliminates any flakiness.